### PR TITLE
Fix an unexpected behavior when chat-closed is received.

### PR DESCRIPTION
**Pull request by jianjunz**

### CODE REVIEW

**Review by taste1981 — APPROVED** [review]

File: talk/owt/sdk/p2p/p2pclient.cc [hunk 1]
```diff
@@ -293,24 +293,40 @@ void P2PClient::OnSignalingMessage(const std::string& message,
         rtc::GetIntFromJsonObject(stop_info, "code", &code);
         rtc::GetStringFromJsonObject(stop_info, "message", &error);
 
+        if (code == kWebrtcIceGatheringPolicyUnsupported) {
+          auto pcc = that->GetPeerConnectionChannel(remote_id);
```
Comment (taste1981) [inline talk/owt/sdk/p2p/p2pclient.cc:297]: Please check validity of  pc channel before you use it.

File: talk/owt/sdk/p2p/p2pclient.cc [hunk 2]
```diff
@@ -293,24 +293,40 @@ void P2PClient::OnSignalingMessage(const std::string& message,
...
+          std::shared_ptr<LocalStream> stream = pcc->GetLatestLocalStream();
+          std::function<void()> success_callback =
+              pcc->GetLatestPublishSuccessCallback();
+          std::function<void(std::unique_ptr<Exception>)> failure_callback =
+              pcc->GetLatestPublishFailureCallback();
+          pcc->SetAbandoned();
+          that->pc_channels_.erase(remote_id);
```
Comment (taste1981) [inline talk/owt/sdk/p2p/p2pclient.cc:304]: Could you clarify why locking is not neccessary for this?

File: talk/owt/sdk/p2p/p2pclient.cc (hunk 1 above)
Comment (jianjunz) [reply talk/owt/sdk/p2p/p2pclient.cc:297]: `GetPeerConnectionChannel` creates a new PeerConnectionChannel if there is no available channel associated with `remote_id`, so `pcc` should always be valid here.

File: talk/owt/sdk/p2p/p2pclient.cc (hunk 2 above)
Comment (jianjunz) [reply talk/owt/sdk/p2p/p2pclient.cc]: Fixed. Locking is required here.

**Review by taste1981 — APPROVED** [review]

**Pull request merged**